src/NUnitFramework/framework/Internal/Execution/ParallelWorkItemDispatcher.cs

```diff
@@ -306,8 +304,12 @@ internal static ExecutionStrategy GetExecutionStrategy(WorkItem work)
                 work.Test is TestFixture && work.Context.ParallelScope.HasFlag(ParallelScope.Fixtures))
                     return ExecutionStrategy.Parallel;
 
-            // If all else fails, run on same thread
-            return ExecutionStrategy.Direct;
+            // There is no scope specified either on the item itself or in the context.
+            // In that case, simple work items are test cases and just run on the same
+            // thread, while composite work items and teardowns are non-parallel.
+            return work is SimpleWorkItem
+                ? ExecutionStrategy.Direct
+                : ExecutionStrategy.NonParallel;
```

Comment:
The only info the dispatcher could push to it that it doesn't have access to already would necessarily be information about the dispatcher.

As in the Test hierarchy, we use the **Composite** pattern here. Normally, when referring to a Test / WorkItem, you treat them all the same. But in the case of Test, it turned out we needed a member `IsSuite` on some occasions. Maybe it's the same here... we might want either `IsSimpleItem` or `IsCompositeItem`, which would have to be implemented by all WorkItems.